Adds login feature to access watchlist and add to watchlist

SwagOtaku opened the pull request:
This pull request adds a login feature as requested in issue #23.

The only issue I'm having is that on my computer the watchlist loads up quickly, roughly 10-20 seconds, while on a firestick the watchlist, it varies from some watchlist folders loading up in 20-30 seconds, while some take up to 1 minute to 1 minute 30 seconds. Still trying to figure out to address this issue. 

If guys have any ideas on how to fix the issue that would be awesome.

pjcanales commented:
Maybe due to the fact the firesticks are not exactly up there in performance wise with your pc. Never count on a firesticks. If you got a shield tv test on that instead.

SwagOtaku commented:
@pjcanales Ok that makes a lot of sense. So maybe it only takes that long on a firestick, unfortunately, I don't have shield.

pjcanales commented:
What's the spec of your computer?

DxCx commented on addon.xml:
```diff
@@ -4,6 +4,7 @@
     <import addon="script.module.requests" version="2.12.4" />
     <import addon="script.common.plugin.cache" version="2.5.5" />
     <import addon="script.module.urlresolver" version="5.0.14" />
+    <import addon="script.module.beautifulsoup4" version="4.5.3"/>  
```
`script.module.beautifulsoup4`
is this served by default kodi?
or we will need to import it to our repo?
also, i havnt seen it was used, if this is a leftover of some test, lets remove it

DxCx commented on default.py:
```diff
@@ -4,9 +4,11 @@
 
 AB_LIST = [".", "0"] + [chr(i) for i in range(ord("A"), ord("Z")+1)]
 MENU_ITEMS = [
+    (control.lang(20100), "watchlist"),
```
please use `30009` instead

DxCx commented on default.py:
```diff
@@ -24,9 +26,42 @@
+try: folder = params['folder']
+except: folder = None
+
+if action == 'login':
```
login class & handling should be part of the Browser module not lib.ui
can you refactor that plz?

UI is more kodi-generic stuff, not specific into a certain website..

DxCx replied:
also, i've written `@route` directive long ago but i think you should be able to use it for the login instead of adding another parsing step

DxCx commented on resources/language/English/strings.xml:
```diff
@@ -2,7 +2,9 @@
 	<string id="20000">Extract Images: </string>
-
+	
+	<string id="20100">Watch List</string>
```
30009 since its used for main menu

DxCx replied:
also use 3020X for the new dialog please

DxCx commented on default.py:
```diff
@@ -53,6 +88,14 @@ def ANIMES_PAGE(animeurl):
+def LOGIN_MENU(payload):
+    if control.getSetting("9anime.username") == '':
+        dialog = xbmcgui.Dialog()
+        dialog.ok("9anime Watch List", "Need to login to access Watch List")
```
lets use localized strings please (control.lang(...))

DxCx commented on resources/lib/ui/BrowserBase.py:
```diff
@@ -21,3 +25,18 @@ def _get_request(self, url, data=None, set_request=None):
             url = "%s?%s" % (url, urllib.urlencode(data))
         return self._send_request(url, None, set_request)
+
+    def _get_watchlist_request(self, url, data=None):
```
should be move into `NineAnimeBrowser` class

DxCx commented on resources/lib/ui/control.py:
```diff
@@ -65,6 +66,21 @@ def xbmc_add_dir(name, url, iconimage='', description=''):
     liz=xbmcgui.ListItem(name, iconImage="DefaultFolder.png", thumbnailImage=iconimage)
     liz.setInfo('video', infoLabels={ "Title": name, "Plot": description })
     liz.setProperty("fanart_image", iconimage)
+    sysaddon = sys.argv[0]
```
we should add those as extra actions only if logged in,
also, this should be done via parameter from `default.py` call

DxCx commented on resources/lib/ui/login.py:
```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
```
please move into `NineAnimeBrowser` also note it has an internal member holding domain,
when we add the feature of changing domain via settings this will be broken..

DxCx commented on default.py:
```diff
@@ -24,9 +26,42 @@
 control.setContent('tvshows');
 
+params = dict(urlparse.parse_qsl(sys.argv[2].replace('?', '')))
+print "params", params
```
leftover print :D

SwagOtaku commented:
Thanks for the feedback @DxCx , I will make the necessary changes you asked for in a couple of days or so.

In regards to your question of script.module.beautifulsoup4, it's an official Kodi addon module so it should automatically download if people don't have it without the need to add to your repo.

One thing I'm confused on is when you say, "we should add those as extra actions only if logged in,
also, this should be done via parameter from default.py call." What do you mean by via parameter from default.py?

DxCx commented:
in `deafult.py` in you have a call to the function to draw the items, just pass the list of extra actions from there, and only if logged in =)
then on control function you get a list to append =)

pjcanales commented:
When will this login feature be introduce.

SwagOtaku commented:
Ok, just tested on firestick with new changes and load time issue is back. I'm gonna try to fix the issue, maybe it is tied with me changing the base_url in NineAnimeBrowser.py

SwagOtaku commented:
I figured out how to fix the issue by decreasing the display number of items in the list through the 9anime website ( https://imgur.com/R47PblU ) and also changing requests.content to requests.text. This loads up the watchlist folders in 10-20 seconds on the firestick.

SwagOtaku commented:
@DxCx I think I found another issue, when I was testing it, the changes made break episode sources

SwagOtaku commented:
I figured out that changing _BASE_URL breaks episode source fetching, so I updated NineAnimeBrowser.py with the original _BASE_URL and added back in the internal member holding domain for now.

DxCx commented:
@SwagOtaku are you done? Can i take it from here?

SwagOtaku commented:
@DxCx Yeah. Go for it 👍

DxCx commented:
Replaced by #245
